### PR TITLE
Add PCAP_OVER_IP support

### LINKED ISSUE #7: PCAP-over-IP support
Hello, and thanks for creating Shovel!

I wanted to open a discussion about potentially adding [PCAP-over-IP](https://www.netresec.com/?page=Blog&month=2022-08&post=What-is-PCAP-over-IP) support to Shovel. As the author of [pcap-broker](https://github.com/fox-it/pcap-broker), another tool in this space, I've come to like using PCAP-over-IP during A/D CTFs wherever possible :)

While Suricata doesn't natively support PCAP-over-IP ([yet](https://redmine.openinfosecfoundation.org/issues/5499)), it's possible to read PCAP data from stdin in combination with netcat until it does.

Adding PCAP-over-IP support would also enhance the ease of setup with docker-compose for acquiring PCAP data remotely and not having to setup extra interfaces.

Is there any interest in supporting this in Shovel? If so, i'm more than happy to draft a PR as i've already successfully used this setup with Shovel before. They are fairly minimal changes to docker-compose.yaml and entrypoint.sh.

### PR DESCRIPTION
**Pull request by yunzheng**
- Added "Option C" example and pcap-broker stub to docker-compose.yml
- Handle $PCAP_OVER_IP environment variable in suricata/entrypoint.sh

Closes #7

### CODE REVIEW

**Comment by yunzheng** [comment]
Some explanation of the changes made in this PR,

Changes to suricata/entrypoint.sh
* `set -o pipefail` ensures that if `nc` disconnects that suricata also exits immediately
* changed to `eval` the suricata command so we don't have to duplicate commandline arguments 
* the `PCAP_OVER_IP` variable uses `tr` to convert `:` to ` ` so we can just define one argument for both hostname and port
* `nc` uses `-d` argument, so netcat doesn't try to read from stdin (it has none).

Changes to docker-compose.yml:
* added `restart: always` to all docker services to ensure it retries reconnecting/restarting
* added `pcap-broker` stub which builds the container using an inline Dockerfile
* the `container_name` is set to `pcap-broker`, which is then used in the "Option C" example in the `PCAP_OVER_IP` environment variable for connecting to it.
* the `PCAP_COMMAND` example uses multiline docker arguments so you can easily change the BPF or commands. The example is based on a faustctf setup.

I'm also happy to update the documentation if needed or feel free to make changes in my PR branch.

**Review by aiooss-anssi — CHANGES_REQUESTED** [review]
Looks good! Please fix the minor changes I requested and I will merge.

I plan to add later some documentation to also show how to pipe a remote interface to a pcap-over-ip endpoint with netcat, as an alternative to pcap-broker (I like the idea of giving to the user multiple choices).

File: docker-compose.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ services:
   suricata:
     build: ./suricata
     image: anssi/shovel-suricata:dev
+    restart: always
```
[the remark this answers is not in the document]
Comment (yunzheng) [reply docker-compose.yml]: That makes sense, thanks for the explaining. I moved `restart: always` to Option C

File: suricata/entrypoint.sh [hunk 2]
```diff
@@ -2,9 +2,18 @@
 # Copyright (C) 2024  ANSSI
 # SPDX-License-Identifier: CC0-1.0
 
+set -euo pipefail
```
[the remark this answers is not in the document]
Comment (yunzheng) [reply suricata/entrypoint.sh:6]: done

**Review by aiooss-anssi — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**